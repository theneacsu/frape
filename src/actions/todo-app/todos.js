import uuidv4 from 'uuid/v4'

const addTodo = text => ({
  type: 'ADD_TODO',
  todo: {
    text,
    id: uuidv4(),
    completed: false
  }
})

const editTodo = (id, updates) => ({
  type: 'EDIT_TODO',
  id,
  updates
})

const removeTodo = id => ({
  type: 'REMOVE_TODO',
  id
})

const toggleTodo = id => ({
  type: 'TOGGLE_TODO',
  id
})

export { addTodo, editTodo, removeTodo, toggleTodo }
