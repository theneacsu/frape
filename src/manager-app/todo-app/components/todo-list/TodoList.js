import React from 'react'
import { connect } from 'react-redux'
import TodoListItem from '../todo-list-item/TodoListItem'
import getVisibleTodos from '../../../../selectors/todo-app/todos'

export const TodoList = props => (
  <div>
    {props.todos.length === 0 ? (
      <p>No todos left.</p>
    ) : (
      <p>
        You have{' '}
        {props.todos.reduce((acc, todo) => (todo.completed ? acc : acc + 1), 0)}{' '}
        todo(s) left.
      </p>
    )}
    {props.todos.map(todo => (
      <TodoListItem key={todo.id} {...todo} />
    ))}
  </div>
)

const mapStateToProps = state => ({
  todos: getVisibleTodos(state.todos, state.todosFilters)
})

export default connect(mapStateToProps)(TodoList)
