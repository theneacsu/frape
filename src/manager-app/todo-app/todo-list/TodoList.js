import React from 'react'
import TodoListItem from './todo-list-item/TodoListItem'

const TodoList = () => (
  <div>
    <TodoListItem />
    <TodoListItem />
    <TodoListItem />
  </div>
)

export default TodoList
