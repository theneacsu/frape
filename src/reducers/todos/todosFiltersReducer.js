const defaultState = {
  searchTerm: '',
  showAll: true,
  hideCompleted: false
}

const todosFiltersReducer = (state = defaultState, action) => {
  switch (action.type) {
    case 'SET_SEARCH_TERM':
      return {
        ...state,
        searchTerm: action.searchTerm
      }
    case 'SET_SHOW_ALL':
      return {
        ...state,
        showAll: true,
        hideCompleted: false
      }
    case 'SET_HIDE_COMPLETED': {
      return {
        ...state,
        showAll: false,
        hideCompleted: true
      }
    }
    default:
      return state
  }
}

export default todosFiltersReducer
