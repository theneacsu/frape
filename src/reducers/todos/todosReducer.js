const todosReducer = (state = [], action) => {
  switch (action.type) {
    case 'ADD_TODO':
      return [...state, action.todo]
    case 'EDIT_TODO':
      return state.map(todo => {
        if (todo.id === action.id) {
          return {
            ...todo,
            ...action.updates
          }
        }
        return todo
      })
    case 'REMOVE_TODO':
      return state.filter(todo => todo.id !== action.id)
    case 'TOGGLE_TODO':
      return state.map(todo => {
        if (todo.id === action.id) {
          return {
            ...todo,
            completed: !todo.completed
          }
        }
        return todo
      })
    default:
      return state
  }
}

export default todosReducer
