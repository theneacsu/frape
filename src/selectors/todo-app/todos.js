const getVisibleTodos = (todos, filters) => {
  const searchTermMatchedTodos = todos.filter(todo =>
    todo.text.toLowerCase().includes(filters.searchTerm.toLowerCase())
  )

  if (filters.hideCompleted) {
    return searchTermMatchedTodos.filter(todo => !todo.completed)
  }

  return searchTermMatchedTodos
}

export const getTodoById = (todos, id) => todos.find(todo => todo.id === id)

export default getVisibleTodos
