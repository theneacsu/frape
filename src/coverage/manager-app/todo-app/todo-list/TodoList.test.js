import React from 'react'
import { shallow } from 'enzyme'
import { TodoList } from '../../../../manager-app/todo-app/todo-list/TodoList'
import todos from '../../../fixtures/todo-app/todos'

test('Should render correctly the TodoList component where there are no todos', () => {
  const wrapper = shallow(<TodoList todos={[]} />)
  expect(wrapper).toMatchSnapshot()
})

test('Should render correctly the TodoList component when there are todos', () => {
  const wrapper = shallow(<TodoList todos={todos} />)
  expect(wrapper).toMatchSnapshot()
})
