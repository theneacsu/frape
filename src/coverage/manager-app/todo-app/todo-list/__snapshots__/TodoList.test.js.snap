// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly the TodoList component when there are todos 1`] = `
<div>
  <p>
    You have
     
    2
     
    todo(s) left.
  </p>
  <Connect(TodoListItem)
    completed={false}
    id="1"
    key="1"
    text="Go for a chest workout"
  />
  <Connect(TodoListItem)
    completed={true}
    id="2"
    key="2"
    text="Learn a new spanish verb"
  />
  <Connect(TodoListItem)
    completed={false}
    id="3"
    key="3"
    text="Talk with Jane about the trip"
  />
</div>
`;

exports[`Should render correctly the TodoList component where there are no todos 1`] = `
<div>
  <p>
    No todos left.
  </p>
</div>
`;
